backend/shared/shared/azure_adfs/auth.py

```diff
@@ -66,6 +66,38 @@ def update_user_groups_from_graph_api(
             new_groups = [Group.objects.get_or_create(name=name)[0] for name in groups]
             user.groups.set(new_groups)
 
+    def get_userinfo_from_graph_api(self, graph_api_access_token: str):
+        """
+        Makes a call to https://docs.microsoft.com/en-us/graph/api/user-get to get more information
+        about the logged in user.
+
+        :returns dictionary of user's requested properties
+        """
+        url = "https://graph.microsoft.com/v1.0/me"
+        properties = (
+            "givenName",
+            "surname",
+        )
+
+        headers = {
+            "Authorization": f"Bearer {graph_api_access_token}",
+        }
+
+        response = requests.get(
+            url, headers=headers, params=f"$select={','.join(properties)}"
+        )
+
+        response.raise_for_status()
+        return response.json()
+
+    def update_userinfo_from_graph_api(self, user, graph_api_access_token: str):
+        userinfo = self.get_userinfo_from_graph_api(graph_api_access_token)
+
+        if "givenName" in userinfo:
+            user.first_name = userinfo["givenName"]
```

Comment:
Is there any scenario where userinfo["givenName"] or userinfo["surname"] is None? Maybe check just in case?